src/http/buf.rs

```diff
@@ -226,6 +227,20 @@ impl MemSlice {
     fn get(&self) -> &[u8] {
         unsafe { &(*self.buf.get())[self.start..self.end] }
     }
+
+    pub fn as_str(&self) -> &str {
```

Comment:
I've been thinking we'd need something like this, but a slightly different way may be better.

```rust
pub struct MemStr(MemSlice);

impl MemStr {
    pub unsafe fn from_utf8_unchecked(slice: MemSlice) -> MemStr {
        MemStr(slice)
    }
}
```

This forces us to use an `unsafe {}` block whenever we create a string, so we don't accidentally just create a `MemStr` that might not be utf8.